pkg/plugins/errors.go

```diff
@@ -35,4 +35,16 @@ var (
 	// Exposed as a base error to wrap it with plugin cancelled errors.
 	ErrPluginRequestCanceledErrorBase = errutil.ClientClosedRequest("plugin.requestCanceled",
 		errutil.WithPublicMessage("Plugin request canceled"))
+
+	// ErrGrpcResourceExhaustedBase error returned when a plugin response is larger than the grpc limit.
+	// Exposed as a base error to wrap it with plugin resource exhausted errors.
+	ErrGrpcResourceExhaustedBase = errutil.Internal("plugin.resourceExhausted",
```

Comment:
Maybe rather than making these gRPC specific, they could be a bit more generic and therefore more reusable for other places? If necessary, we could put extra context (like gRPC information) when forming the error for example

Reply:
Yeah, that is a good feedback. Definitely!